src/AssetScanner.php

```diff
@@ -90,19 +95,24 @@ protected function _expandPaths()
     }
 
     /**
-     * Discover all the sub directories for a given path.
+     * Get an array of all the folders inside directory structure recursively.
      *
-     * @param string $path The path to search
-     * @return array Array of subdirectories.
+     * @param string $dir The path of root directory to look through.
+     * @param array $result The array of folder paths.
+     * @return void
      */
-    protected function _generateTree($path)
+    protected function _dirToArray($dir, &$result)
```

Comment:
Pass by reference makes the code harder to follow and is a pattern I think should be avoided. PHP internals do all sorts of silly things that userland code generally avoids.